addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -251,7 +251,7 @@ def _compute_user(self):
         for applicant in self:
             applicant.user_id = applicant.job_id.user_id.id or self.env.uid
 
-    @api.depends('partner_id')
+    @api.depends('partner_id', 'partner_id.phone', 'partner_id.mobile', 'partner_id.email')
```

Comment:
@bedo @jugj-odoo 

Actually not :smile: 

I know I know, it seems logical on the first glance, but it was deliberate to have it like this. Here are two fixes that explain why it is so. 
https://github.com/odoo/odoo/commit/72c62dbf157dfb353caf13a228921e3f81036885
https://github.com/odoo/odoo/commit/264db553635048cd94dc5a8c5f608fe844099c60

@jugj-odoo, so please undo the change :innocent: 

Now, that's being said, maybe we could add a little bit of a explanatory comment so that people who stumble across this code are not confused?

"""
Note, we want to prevent synchronization between applicant
and partner if partner information changes.
The partner must be changed to update the applicant.
"""

Reply:
also the commit header :bulb: